lib/AST/Decl.cpp

```diff
@@ -2584,6 +2584,13 @@ ObjCClassKind ClassDecl::checkObjCAncestry() const {
   return kind;
 }
 
+ClassDecl::MetaclassKind ClassDecl::getMetaclassKind() const {
+  assert(getASTContext().LangOpts.EnableObjCInterop &&
+         "querying metaclass kind without objc interop");
+  auto objc = checkObjCAncestry() != ObjCClassKind::NonObjC || hasClangNode();
```

Comment:
I don't know. Based on the blame it seems some person called @slavapestov was the most recent to touch this (in https://github.com/apple/swift/commit/871d71a555ddeb5a572cc90ea8415b782c710db6 ), but the logic has been here since @rjmccall made the compiler decide between objc and swift metaclasses more than 4 years ago. I'll try removing it.